### PR TITLE
Update and Repaint on resize.

### PR DESCRIPTION
**Pull request by otac0n**
# Summary of the PR
Repaint on resize/drag.
This does NOT automatically queue paint messages with a timer or the like, so the view still hangs as long as the mouse is held still.

# Related issues, Discord discussions, or proposals
https://github.com/dotnet/Silk.NET/issues/241

# Further Comments
I am not sure how (or if) to implement this for the SdlView.

### CODE REVIEW

File: src/Windowing/Silk.NET.Windowing.Glfw/GlfwWindow.cs
Comment (otac0n) [inline src/Windowing/Silk.NET.Windowing.Glfw/GlfwWindow.cs:664]: Another feasible option is to keep the API surface the same and call the  Update/Render methods here, perhaps with a zero update value to indicate an inter-frame update.

**Comment by Perksey** [comment]
This has been added to the Working Group Meeting agenda so that we can ratify API changes. Realistically this is minimal, but there are other API changes we have to discuss that we can sweep this under. No date set yet, I have one more proposal to write before we can set one, but hopefully it shouldn't be too long.

**Comment by otac0n** [comment]
https://github.com/dotnet/Silk.NET/assets/193584/a2f2f9c0-c14a-42c9-b945-003529426cdd

File: src/Windowing/Silk.NET.Windowing.Common/WindowExtensions.cs
Comment (Perksey) [inline src/Windowing/Silk.NET.Windowing.Common/WindowExtensions.cs:66]: I appreciate the intent, but I don't think automatically binding to events is good. If we want this to be default behaviour, then maybe we should just make it so. Wouldn't be the worst idea actually, then we wouldn't need Working Group approval at all and we can sweep the SDL vs GLFW behaviour differences under the rug as "undefined behaviour" (seeing as we only officially support SDL on Android and experimentally on iOS)

File: src/Windowing/Silk.NET.Windowing.Glfw/GlfwWindow.cs
Comment (Perksey) [reply src/Windowing/Silk.NET.Windowing.Glfw/GlfwWindow.cs:664]: I think I'm leaning more towards this after seeing the latest changes (per the above comment)

**Review by Perksey — CHANGES_REQUESTED** [review]
Pre-WG comments

File: src/Windowing/Silk.NET.Windowing.Glfw/GlfwWindow.cs
Comment (Perksey) [inline src/Windowing/Silk.NET.Windowing.Glfw/GlfwWindow.cs:672]: This is fine, and I'm pretty happy to just merge as is, but one possible Q which I'm not sure how much we should care about is "what if they're not using our Render/Update events?"

What we could do:
- Have a boolean field on the window e.g. `_isInDoEvents` that is set to `true` when `DoEvents` is called.
- If `DoEvents` is called and this is already `true`, `DoEvents` does not execute
- In addition to the `Run` implementation just being a `while (true)` loop by default, we could capture the `Action` passed in and use that for `_onRefresh` instead.

The reason why we need the boolean flag is because the user's action will naturally try to call `DoEvents`, which would cause infinite recursion/undefined behaviour.

So sorry I didn't think of this earlier, I really don't want to mess you around with this too much. This is technically fine so we can merge this as is if you want, but I just remembered that this is exactly the sort of scenario that `Run(Action)` was designed for!

**Review by Perksey — COMMENTED** [review]
Just a quick one your thoughts would be appreciated on...

File: src/Windowing/Silk.NET.Windowing.Glfw/GlfwWindow.cs
Comment (otac0n) [reply src/Windowing/Silk.NET.Windowing.Glfw/GlfwWindow.cs:672]: Can you please give me a test case for such an Update() event?  I am not 100% sure I follow and I want to make sure I cover your concern.

**Review by Perksey — APPROVED** [review]
Perfect, thank you!

**Pull request merged**
